src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -31,7 +31,8 @@
     public static Index parseIndex(String oneBasedIndex) throws ParseException {
         String trimmedIndex = oneBasedIndex.trim();
         if (!StringUtil.isNonZeroUnsignedInteger(trimmedIndex)) {
-            throw new ParseException(MESSAGE_INVALID_INDEX);
+            throw new ParseException(MESSAGE_INVALID_INDEX
+                + " Given trimmed index: [" + trimmedIndex + "]");
```

Comment:
Need a full stop in between `MESSAGE_INVALID_INDEX` and the new addition, and another at the end.
Also have a message like this to be more user-friendly:
`throw new ParseException(MESSAGE_INVALID_INDEX + ". You entered [" + trimmedIndex + "] for the index.");`
and similar messages for the others below.

Reply:
Actually, I'm thinking maybe revert this change. It's causing a lot of regression tests to fail since these tests use only the static private final error messages for validating a correctly failing test.

If we want this user-friendly error message change, we might need a more general "create error message" function. 

This isn't an urgent feature, so I'm in favor of reverting the commit and merging the other changes first. What do you think?

Reply:
> Actually, I'm thinking maybe revert this change. It's causing a lot of regression tests to fail since these tests use only the static private final error messages for validating a correctly failing test.
> 
> If we want this user-friendly error message change, we might need a more general "create error message" function.
> 
> This isn't an urgent feature, so I'm in favor of reverting the commit and merging the other changes first. What do you think?

Actually, now that I actually look at it, it's only a few tests here and there. In this case, maybe we could have a new method in each of the field classes like ```formErrorMessage(String input)```